recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+        )
+        os.rename(extracted_dir, self._source_subfolder)
+        self._patch_sources()
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_ssl == 'OPENSSL':
+            self.requires("openssl/1.1.1g")
+        if self.options.with_ssl == 'LIBRESSL':
+            self.output.warn("Can be broken. Prefer OpenSSL instead of LIBRESSL")
+            self.requires("libressl/3.2.0")
+        if self.options.with_zlib == 'SYSTEM':
+            self.requires("zlib/1.2.11")
+            if tools.os_info.is_windows:
+                self.output.warn("Usage zlib provided by conan on Windows can be broken")
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD_PARTY_NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["bson-1.0", "mongoc-1.0"] if self.options.shared else \
+            ["bson-static-1.0", "mongoc-static-1.0"]
+        self.cpp_info.includedirs.append(os.path.join("include", "libmongoc-1.0"))
+        self.cpp_info.includedirs.append(os.path.join("include", "libbson-1.0"))
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = "TRUE"
+        self._cmake.definitions["ENABLE_TESTS"] = "OFF"
+        self._cmake.definitions["ENABLE_EXAMPLES"] = "OFF"
+        self._cmake.definitions["ENABLE_STATIC"] = "OFF" if self.options.shared else "ON"
+        self._cmake.definitions["ENABLE_SSL"] = self.options.with_ssl
+        self._cmake.definitions["ENABLE_ZLIB"] = self.options.with_zlib
+        self._cmake.definitions["ENABLE_SHM_COUNTERS"] = "OFF"
+        self._cmake.definitions["ENABLE_BSON"] = "ON"
+        self._cmake.definitions["ENABLE_MONGOC"] = "ON"
+        if self.settings.os == "Windows":
+            self._cmake.definitions["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
+        if self.options.with_zlib == 'SYSTEM':
+            self._cmake.definitions["ZLIB_ROOT"] = self.deps_cpp_info["zlib"].rootpath
+        if self.options.with_ssl == 'OPENSSL':
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath
```

Comment:
There are also several others options in upstream CMakeLists which shouldn't be left to AUTO value (ENABLE_SNAPPY and ENABLE_ZSTD for example).

Reply:
addressed in n-bes#1